industry_benchmarks/utils/data_gathering.py

```diff
@@ -444,14 +466,247 @@ def gather_ligand_scores(
+
+    # Check to make sure we don't just have failures
+    # if all failures, tell user to re-run
+    if protocol_unit_result_count == 0:
+        print(f"{file_name} failed to run, traceback and exception below, this will not be included in the results. \n")
+        proto_failures = [
+            k
+            for k in results["unit_results"].keys()
+            if k.startswith("ProtocolUnitFailure")
+        ]
+        for proto_failure in proto_failures:
+            print("\n")
+            print(results["unit_results"][proto_failure]["traceback"])
+            print(results["unit_results"][proto_failure]["exception"])
+            print("\n")
+        return None
+
+    return results
+
+def remove_first_reversed_sequential_duplicate_from_path(path: pathlib.Path) -> pathlib.Path:
+    """
+    Remove the first duplicated directory from path
+    We reverse the path so we remove the first sequential
+    duplicated directory starting from the deepest part of the path
+    """
+
+    # reverse the path parts so we start from the deepest part first
+    reversed_path_parts = list(reversed(path.parts))
+    max_idx = len(reversed_path_parts)
+
+    # find index of first dupe
+    for idx in range(max_idx - 1):
+        if reversed_path_parts[idx] == reversed_path_parts[idx + 1]:
+            break
+    # no dupes
+    else:
+        print("Path didn't have any dupes")
+        return path
+
+    del reversed_path_parts[idx]
+    return pathlib.Path(*reversed(reversed_path_parts))
+
+
+def find_data_folder(result: dict) -> None | pathlib.Path:
+    """
+    Find the path to the cleaned up results file for this transformation result.
+    """
+    # get the name of the key which is a gufe token
+    # for the only ProtocolUnitResult-* in unit_results
+    # this means we can grab the first that matches since there is only
+    # one ProtocolUnitResult-*
+    proto_key = next(
+        k
+        for k in result["unit_results"].keys()
+        if k.startswith("ProtocolUnitResult")
+    )
+    results_dir = (
+        pathlib.Path(result["unit_results"][proto_key]["outputs"]["nc"]["path"])
+        .resolve()
+        .parent
+    )
+    # if the dir doesn't exist, we should try and fix it
+    if not results_dir.is_dir():
+        print("Fixing path to results dir")
+        # Depending on the relative location to the result dir, we might have
+        # to fix a duplicate folder, see this post for more details
+        # https://github.com/OpenFreeEnergy/IndustryBenchmarks2024/pull/83#discussion_r1689003616
+        results_dir = remove_first_reversed_sequential_duplicate_from_path(
+            results_dir
+        )
+        # Now we should check if the dir exists
+        if not results_dir.is_dir():
+            error_message = f"Can't find results directory: {results_dir}"
+            raise FileNotFoundError(error_message)
+
+    # now check that all of the results files can be found in the folder
+    for f_name in RESULT_FILES:
```

Comment:
Maybe for the png file (esp. the "forward_reverse_convergence.png") we could just print a warning, but not fail? There sometimes can be a problem with the forward backward convergence analysis because of the way how we do autocorrelation. When just using 10% of the data the analysis can fail and we don't return the png.

Reply:
I allowed the skipping of all png files if they can not be found for now, do you think that would be okay?

Reply:
Yes, I think that should be ok!

Reply:
That works, we'll likely re-calculate all the energy PNGs after the fact anyways (with pymbar 4), since we know the overlap matrix can be odd with pymbar 3.